packages/@react-spectrum/cards/test/CardView.test.js

```diff
@@ -0,0 +1,1139 @@
+            columnHeights.push(expectedHeight);
+          } else {
+            let index = columnLefts.indexOf(div.style.left);
+            expect(index).not.toEqual(-1);
+            expect(columnHeights[index]).toEqual(div.style.top);
+            columnHeights[index] = `${parseInt(div.style.top, 10) + parseInt(div.style.height, 10) + 24}px`;
+          }
+        }
+      }
+    });
+
+    describe('keyboard nav', function () {
+      it('should move focus via Arrow Down', function () {
+        let tree = render(<DynamicCardView layout={WaterfallLayout} />);
+        act(() => {
+          jest.runAllTimers();
+        });
+
+        let cards = tree.getAllByRole('gridcell');
+        triggerPress(cards[0]);
+        expect(document.activeElement).toBe(cards[0]);
+        expect(within(document.activeElement).getByText('Title 1')).toBeTruthy();
+
+        let cardStyles = getCardStyles(cards[0]);
+        let expectedTop = `${parseInt(cardStyles.top, 10) + parseInt(cardStyles.height, 10) + 24}px`;
+        let expectedLeft = cardStyles.left;
+
+        act(() => {
+          fireEvent.keyDown(document.activeElement, {key: 'ArrowDown', code: 40, charCode: 40});
+          fireEvent.keyUp(document.activeElement, {key: 'ArrowDown', code: 40, charCode: 40});
+          jest.runAllTimers();
+        });
+
+        cardStyles = getCardStyles(document.activeElement);
+        expect(cardStyles.top).toEqual(expectedTop);
+        expect(cardStyles.left).toEqual(expectedLeft);
+        expect(within(document.activeElement).getByText('Title 4')).toBeTruthy();
+      });
+
+      it('should move focus via Arrow Up', function () {
+        let tree = render(<DynamicCardView layout={WaterfallLayout} />);
+        act(() => {
+          jest.runAllTimers();
+        });
+
+        let cards = tree.getAllByRole('gridcell');
+        triggerPress(cards[3]);
+        expect(document.activeElement).toBe(cards[3]);
+        expect(within(document.activeElement).getByText('Title 4')).toBeTruthy();
+
+        let cardStyles = getCardStyles(cards[3]);
+        let expectedTop = cardStyles.top;
+        let expectedLeft = cardStyles.left;
+
+        act(() => {
+          fireEvent.keyDown(document.activeElement, {key: 'ArrowUp', code: 38, charCode: 38});
+          fireEvent.keyUp(document.activeElement, {key: 'ArrowUp', code: 38, charCode: 38});
+          jest.runAllTimers();
+        });
+
+        cardStyles = getCardStyles(document.activeElement);
+        expectedTop = `${parseInt(expectedTop, 10) - parseInt(cardStyles.height, 10) - 24}px`;
+        expect(cardStyles.top).toEqual(expectedTop);
+        expect(cardStyles.left).toEqual(expectedLeft);
+        expect(within(document.activeElement).getByText('Title 1')).toBeTruthy();
+      });
+
+      // TODO: figure out why the spacing is only 16px between each item
+      it.skip('should move focus via Arrow Left', function () {
```

Comment:
? have you figured it out? or want help?

Reply:
Will need to investigate some more later, might be fixed in some of the follow up PRs